### PR TITLE
Create auth methods and user creation methods

### PR DESCRIPTION
**Pull request by ShinUrech**
Firebase 
- Initialize firebase in the project: root/firebase/firebase.tsx root/.firebasesrc
- Initialize firebase emulators for auth
- Initialize firebase auth and firestore: root/firebase/auth/auth.ts and root/firebase/firestore/firestore.ts
- Install firebaseui

Methods linked to authentication: root/firebase/auth/auth.ts
- login methods with google auth provider and email and password 
- sign in methods with googe and email and password
- account email modification for the account
- password change method
- account sign out method
- account deletion method 
JEST unit-tests for methods above: root/__tests__/auth.tests.ts

### CODE REVIEW

File: ui-debug.log [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    throw new ERR_REQUIRE_ESM(filename, true);
+    ^
+
+Error [ERR_REQUIRE_ESM]: require() of ES Module /home/shin/.cache/firebase/emulators/ui-v1.14.0/server/server.mjs not supported.
+Instead change the require of /home/shin/.cache/firebase/emulators/ui-v1.14.0/server/server.mjs to a dynamic import() which is available in all CommonJS modules.
+    at Function.runMain (pkg/prelude/bootstrap.js:1979:12) {
+  code: 'ERR_REQUIRE_ESM'
+}
+
+Node.js v18.5.0
```
Comment (AdrienDeschenaux) [inline ui-debug.log:11]: Does this file need to be pushed ? I feel that this file shouldn't be pushed, let me know if it should be merged and if you could explain me why that would be great.

Comment (AdrienDeschenaux) [reply ui-debug.log]: Thank you for your feedback we'll check this in order to merge your branch soon

File: firestore-debug.log
Comment (userdarius) [inline firestore-debug.log:1]: I don't think we need this if we could add it to the gitignore it would be best.

File: package.json
Comment (userdarius) [inline package.json:1]: We should not use the react test renderer lib

File: __tests__/auth.test.ts
Comment (userdarius) [inline __tests__/auth.test.ts:1]: LGTM

File: firebase/auth/auth.ts
Comment (userdarius) [inline firebase/auth/auth.ts:1]: LGTM

File: firebase/firebase.ts
Comment (userdarius) [inline firebase/firebase.ts:1]: Looks good although (nitpicking) we could simplify the export to keep the code more concise

**Review by userdarius — CHANGES_REQUESTED** [review]
Clean PR some review needed by @ShinUrech for the package dependencies

**Review by userdarius — APPROVED** [review]

**Pull request merged**